src/privileged/FirefoxMonitor.jsm

```diff
@@ -0,0 +1,353 @@
+
+  getURL(aPath) {
+    return this.extension.getURL(aPath);
+  },
+
+  getString(aKey) {
+    return this.strings.GetStringFromName(aKey);
+  },
+
+  getFormattedString(aKey, args) {
+    return this.strings.formatStringFromName(aKey, args, args.length);
+  },
+
+  async init(aExtension) {
+    this.extension = aExtension;
+
+    XPCOMUtils.defineLazyPreferenceGetter(
+      this, "enabled", this.kEnabledPref, false,
+      async (pref, oldVal, newVal) => {
+        if (newVal) {
+          if (!this._delayedInited) {
+            await this.delayedInit();
+          }
+          this.startObserving();
+        } else {
+          this.stopObserving();
+        }
+      }
+    );
+
+    if (!this.enabled) {
+      return;
+    }
+
+    await this.delayedInit();
+    this.startObserving();
+  },
+
+  async delayedInit() {
+    /* globals AddonManager, Preferences, fetch, btoa, XUL_NS */
+    Services.scriptloader.loadSubScript(
+      this.getURL("privileged/subscripts/Globals.jsm"));
+
+    /* globals EveryWindow */
+    Services.scriptloader.loadSubScript(
+      this.getURL("privileged/subscripts/EveryWindow.jsm"));
+
+    /* globals PanelUI */
+    Services.scriptloader.loadSubScript(
+      this.getURL("privileged/subscripts/PanelUI.jsm"));
+
+    Services.telemetry.registerScalars("fxmonitor", {
+      "doorhanger_shown": {
+        kind: Ci.nsITelemetry.SCALAR_TYPE_COUNT,
+        keyed: false,
+        record_on_release: true,
+      },
+      "doorhanger_removed": {
+        kind: Ci.nsITelemetry.SCALAR_TYPE_COUNT,
+        keyed: false,
+        record_on_release: true,
+      },
+      "check_btn_clicked": {
+        kind: Ci.nsITelemetry.SCALAR_TYPE_COUNT,
+        keyed: false,
+        record_on_release: true,
+      },
+      "dismiss_btn_clicked": {
+        kind: Ci.nsITelemetry.SCALAR_TYPE_COUNT,
+        keyed: false,
+        record_on_release: true,
+      },
+      "never_show_btn_clicked": {
+        kind: Ci.nsITelemetry.SCALAR_TYPE_COUNT,
+        keyed: false,
+        record_on_release: true,
+      },
+    });
+
+    let warnedHostsJSON = Preferences.get(this.kWarnedHostsPref, "");
+    if (warnedHostsJSON) {
+      try {
+        let json = JSON.parse(warnedHostsJSON);
+        this.warnedHostsSet = new Set(json);
+      } catch (ex) {
+        // Invalid JSON, invalidate the pref.
+        Preferences.set(this.kWarnedHostsPref, "");
+      }
+    }
+
+    await this.loadStrings();
+    await this.loadBreaches();
+
+    AddonManager.addAddonListener(this);
+
+    this._delayedInited = true;
+  },
+
+  onUninstalled(aAddon) {
+    if (aAddon.id === this.extension.id) {
+      this.stopObserving();
+    }
+  },
+
+  async loadStrings() {
+    // Services.strings.createBundle has a whitelist of URL schemes that it
+    // accepts. moz-extension: is not one of them, so we work around that
+    // by reading the file manually and creating a data: URL (allowed).
+    // TODO:
+    // - Check locale and load relevant file
```

Comment:
I'm a bit uncertain about this TODO item, what's your plan for rollout? I'd say it would be good to resolve this to avoid blocking on it later when localization arrives.